integration/test/app/simple_test.go

```diff
@@ -0,0 +1,203 @@
+					Type: "helm",
+					URL:  "https://giantswarm.github.com/sample-catalog",
+				},
+			},
+			AppOperator: chartvalues.APIExtensionsAppE2EConfigAppOperator{
+				Version: "1.0.0",
+			},
+			Namespace: giantswarm,
+		}
+
+		chartValues, err := chartvalues.NewAPIExtensionsAppE2E(c)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		chartInfo := release.NewStableChartInfo(key.CustomResourceReleaseName())
+		err = config.Release.Install(ctx, key.CustomResourceReleaseName(), chartInfo, chartValues)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		err = config.Release.WaitForStatus(ctx, fmt.Sprintf("%s-%s", giantswarm, key.CustomResourceReleaseName()), "DEPLOYED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created app %#q", key.CustomResourceReleaseName()))
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chart CR %#q is deployed", key.TestAppReleaseName()))
+
+		tarballURL := "https://giantswarm.github.com/sample-catalog/test-app-1.0.0.tgz"
+		operation := func() error {
+			chart, err := config.Host.G8sClient().ApplicationV1alpha1().Charts(giantswarm).Get(key.TestAppReleaseName(), v1.GetOptions{})
+			if err != nil {
+				return microerror.Maskf(err, fmt.Sprintf("expected %#v got %#v", nil, err))
+			}
+			if !reflect.DeepEqual(chart.Spec.TarballURL, tarballURL) {
+				return microerror.Maskf(notMatching, fmt.Sprintf("expected tarballURL: %#v got %#v", tarballURL, chart.Spec.TarballURL))
+			}
+			return nil
+		}
+		notify := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to get created chart: retrying in %s", t))
+		}
+		b := backoff.NewExponential(30*time.Second, 10*time.Second)
+		err = backoff.RetryNotify(operation, b, notify)
+		if err != nil {
+			t.Fatalf("%s", err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("chart %#q is deployed", key.TestAppReleaseName()))
+	}
+
+	// Test update
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating app %#q", key.CustomResourceReleaseName()))
+
+		c := chartvalues.APIExtensionsAppE2EConfig{
+			App: chartvalues.APIExtensionsAppE2EConfigApp{
+				Name:      key.TestAppReleaseName(),
+				Namespace: giantswarm,
+				Catalog:   key.TestAppCatalogReleaseName(),
+				Version:   "1.0.1",
+			},
+			AppCatalog: chartvalues.APIExtensionsAppE2EConfigAppCatalog{
+				Name:  key.TestAppCatalogReleaseName(),
+				Title: key.TestAppCatalogReleaseName(),
+				Storage: chartvalues.APIExtensionsAppE2EConfigAppCatalogStorage{
+					Type: "helm",
+					URL:  "https://giantswarm.github.com/sample-catalog_1/",
+				},
+			},
+			AppOperator: chartvalues.APIExtensionsAppE2EConfigAppOperator{
+				Version: "1.0.0",
+			},
+			Namespace: giantswarm,
+		}
```

Comment:
Can we reuse this object, mutate it and create new chartValues? That would be easier to follow what actually changed in the app CR.